File: src/lib/src/main/java/work/lclpnet/ap2/api/util/scoreboard/InformativeScoreboard.java
package work.lclpnet.ap2.api.util.scoreboard;

import net.minecraft.network.chat.Component;
import work.lclpnet.ap2.impl.util.scoreboard.ScoreHandle;
import work.lclpnet.ap2.impl.util.scoreboard.ScoreboardLayout;
import work.lclpnet.kibu.translate.text.TranslatedText;

public interface InformativeScoreboard {

    ScoreHandle createText(Component text, int position);

    ScoreHandle createText(TranslatedText text, int position);

    default ScoreHandle createText(Component text) {
        return createText(text, ScoreboardLayout.TOP);
    }

    default ScoreHandle createText(TranslatedText text) {
        return createText(text, ScoreboardLayout.TOP);
    }

    default void createNewline(int position) {
        createText(Component.empty(), position);
    }
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/util/scoreboard/InformativeScoreboard.kt
package work.lclpnet.ap2.util.scoreboard

import net.minecraft.network.chat.Component
import work.lclpnet.kibu.translate.text.TranslatedText

interface InformativeScoreboard {

    fun createText(text: Component, position: Int): ScoreHandle

    fun createText(text: TranslatedText, position: Int): ScoreHandle

    fun createText(text: Component): ScoreHandle =
        createText(text, ScoreboardLayout.TOP)

    fun createText(text: TranslatedText): ScoreHandle =
        createText(text, ScoreboardLayout.TOP)

    fun createNewline(position: Int) {
        createText(Component.empty(), position)
    }
}
